source/common/router/config_impl.cc

```diff
@@ -634,7 +634,7 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
       throw EnvoyException(absl::StrCat("Duplicate upgrade ", upgrade_config.upgrade_type()));
     }
     if (upgrade_config.upgrade_type() == Http::Headers::get().MethodValues.Connect) {
-      connect_config_ = upgrade_config.connect_config();
+      connect_config_ = std::make_unique<ConnectConfig>(upgrade_config.connect_config());
```

Comment:
I was wondering whether this can be further optimized by surrounding with an `if (upgrade_config.has_value())`, but looking at the for loop around this whole thing I think that at the moment if there are multiple upgrade_configs and the last hasn't got a connect_config, then the `connect_config_` will be set as empty.

Reply:
Not sure I follow. This seems to be finding the only Connect config, I didn't change that semantics. It throws out of the loop if it finds duplicate types (so only one Connect is possible), and if no Connects are present, then it will remain empty.  Not sure what you mean by `upgrade_config.has_value()` check.

Reply:
Ah, I see, thanks for the clarification. That is indeed orthogonal, and I assume a default ConnectConfig would do in this case (but no idea really). The connect_config field in the proto is documented as "Note that CONNECT support is currently considered alpha in Envoy".